src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -937,14 +937,35 @@ void reorder_inputs::run(program& p, layout_optimizer& lo, reorder_factory& rf)
         }
     };
 
+    const auto reorder_input_concat = [&p, &rf](typed_program_node<concatenation>& concat_node) {
+        auto output_layout = concat_node.get_output_layout();
+        // Iterate over all dependencies of the concat node
+        for (size_t i = 0; i < concat_node.get_dependencies().size(); ++i) {
+            auto dep = concat_node.get_dependency_with_port(i);
+            const auto& input = dep.first;
+            auto input_layout = input->get_output_layout();
+            // Change input data type of concat node from input format to output format
+            if (input_layout.format != output_layout.format) {
+                auto new_layout = input_layout;
+                new_layout.format = output_layout.format;
+                auto new_input = rf.get_reorder(input->id(), dep.second, input_layout, new_layout);
+                if (new_input.first) {
+                    p.add_intermediate(new_input.first, concat_node, i);
+                    concat_node.get_dependency_with_port(i).first->recalc_output_layout();
+                }
+            }
+        }
+    };
+
     for (auto& prim : p.get_processing_order()) {
-        program_helpers::do_for_types<detection_output, deconvolution, convolution, fully_connected, pooling>(
+        program_helpers::do_for_types<detection_output, deconvolution, convolution, fully_connected, pooling, concatenation>(
             *prim,
             reorder_input_detection_output,
             reorder_input_and_weights_deconvolution,
             reorder_convolution,
             reorder_input_fully_connected,
-            reorder_input_pooling);
+            reorder_input_pooling,
+            reorder_input_concat);
```

Comment:
Hmm not sure why this is happening.. Can I ask you some questions? 
* when/why is there case where input layouts of concats are different from output layout?
* IIRC, the case should have been handled from `minimize_local_reorders` function. Why didn't it work?

Reply:
* Other formats seem to be handled properly. And only `Gather_106` (fsv32 input to `Concat_94`) was not. It was bfyx, but after `insert_reorders()`, and it was converted to fsv32 by `recalc_output_layouts()`. And `Concat_94` had no chance to insert a reorder for it.
And this PR tried to insert a reorder using helper function.

* In `minimize_local_reorders`, `Concat_94` had no `best_reorder_cnt.number`, so the other processing was not executed.

Thanks for your comment.